packages/manager/src/auth/PrismicAuthManager.ts

```diff
@@ -0,0 +1,403 @@
+export type PrismicUserProfile = t.TypeOf<typeof PrismicUserProfile>;
+
+type PrismicAuthManagerConstructorArgs = {
+	scopedDirectory?: string;
+};
+
+type PrismicAuthManagerLoginArgs = {
+	email: string;
+	cookies: string[];
+};
+
+type PrismicAuthManagerGetLoginSessionInfoReturnType = {
+	port: number;
+	url: string;
+};
+
+type PrismicAuthManagerNodeLoginSessionArgs = {
+	port: number;
+	onListenCallback?: () => void;
+};
+
+type GetProfileForAuthenticationTokenArgs = {
+	authenticationToken: string;
+};
+
+const checkHasAuthenticationToken = (
+	authState: PrismicAuthState,
+): authState is PrismicAuthState & {
+	cookies: Required<
+		Pick<
+			PrismicAuthState["cookies"],
+			typeof AUTH_COOKIE_KEY | typeof SESSION_COOKIE_KEY
+		>
+	>;
+} => {
+	return Boolean(
+		authState.cookies[AUTH_COOKIE_KEY] && authState.cookies[SESSION_COOKIE_KEY],
+	);
+};
+
+const parseCookies = (cookies: string): Record<string, string> => {
+	return cookie.parse(cookies, {
+		// Don't escape any values.
+		decode: (value) => value,
+	});
+};
+
+export class PrismicAuthManager {
+	// TODO: Automatically scope the manager to the current Slice Machine
+	// project? If not, this internal state can be removed.
+	scopedDirectory: string;
+
+	constructor({
+		scopedDirectory = os.homedir(),
+	}: PrismicAuthManagerConstructorArgs = {}) {
+		this.scopedDirectory = scopedDirectory;
+	}
+
+	// TODO: Make the `cookies` argument more explicit. What are these
+	// mysterious cookies?
+	async login(args: PrismicAuthManagerLoginArgs): Promise<void> {
+		const authState = await this._readPersistedAuthState();
+
+		// Set the auth's URL base to the current base at runtime.
+		authState.base = API_ENDPOINTS.PrismicWroom;
+		authState.cookies = {
+			...authState.cookies,
+			...parseCookies(args.cookies.join(COOKIE_SEPARATOR)),
+		};
+
+		if (checkHasAuthenticationToken(authState)) {
+			const authenticationToken = authState.cookies[AUTH_COOKIE_KEY];
+			const profile = await this._getProfileForAuthenticationToken({
+				authenticationToken,
+			});
+
+			authState.shortId = profile.shortId;
+			authState.intercomHash = profile.intercomHash;
+		}
+
+		await this._writePersistedAuthState(authState);
+	}
+
+	async getLoginSessionInfo(): Promise<PrismicAuthManagerGetLoginSessionInfoReturnType> {
+		// Pick a random port, with a preference for historic `5555`
+		const port = await getPort({ port: 5555 });
+
+		const url = new URL(
+			`./dashboard/cli/login?source=slice-machine&port=${port}`,
+			API_ENDPOINTS.PrismicWroom,
+		).toString();
+
+		return {
+			port,
+			url,
+		};
+	}
+
+	async nodeLoginSession(
+		args: PrismicAuthManagerNodeLoginSessionArgs,
+	): Promise<void> {
+		return new Promise<void>(async (resolve) => {
+			// Timeout attempt after 3 minutes
+			const timeout = setTimeout(() => {
+				server.close();
+				throw new Error(
+					"Login timeout, server did not receive a response within a 3-minute delay",
+				);
+			}, 180_000);
+
+			const app = h3.createApp();
```

Comment:
It should be compatible since the resulting server is a Node.js HTTP server (`http.Server`).

I personally prefer using less abstractions. If we need to test an HTTP endpoint, we can use `fetch()` and read the `Response` object rather than learning Supertest's APIs. I think it's more future proof as the Node.js and JavaScript ecosystem moves toward standard Web APIs.